core/src/test/java/org/apache/iceberg/hadoop/TestHadoopTables.java

```diff
@@ -92,27 +83,23 @@ public void testDropTableWithPurge() throws IOException {
         .isInstanceOf(NoSuchTableException.class)
         .hasMessageStartingWith("Table does not exist");
 
-    Assert.assertEquals(0, dataDir.listFiles().length);
-    Assert.assertFalse(tableDir.exists());
-
-    Assert.assertFalse(TABLES.dropTable(tableDir.toURI().toString()));
+    Assertions.assertThat(Files.list(dataDir.toPath()).count()).isEqualTo(0);
```

Comment:
given that we switched back dataDir to a File, we don't need to call `Files.list(..)` here right?

Reply:
Your review is correct. I'll modify the code now.